hix/default.nix

```diff
@@ -18,6 +18,7 @@ let
     nix-env -f ${hixProject} ${args} "$@"
   '';
   hix = pkgs.writeScriptBin "hix" ''
+      #! /usr/bin/env bash
```

Comment:
I wonder if we should use `writeShellScript` to add the bang pattern?

Reply:
I don't understand exactly why (seems that the default shell here isn't `bash`), but your code change suggestion re-break `hix` tests https://github.com/input-output-hk/haskell.nix/actions/runs/3582998889/jobs/6027893720 + https://input-output-rnd.slack.com/archives/CG386Q1NE/p1669804512458809